sound/soc/sof/ipc4-mtrace.c

```diff
@@ -638,6 +678,10 @@ int sof_ipc4_mtrace_update_pos(struct snd_sof_dev *sdev, int core)
 
 	wake_up(&core_data->trace_sleep);
 
+	/* poll for a while in case DSP sends more data */
+	core_data->poller_idle = 0;
+	mod_delayed_work(system_wq, &core_data->poller_work, 1);
```

Comment:
This is going to run a 100 x 1 jiffies poll run even if the mtrace has not been opened by the user and the firmware happens to send messages?
Should this be done only if the mtrace is in use?